### PR TITLE
Cpfed 3809 pfe nav add overlay

### PR DESCRIPTION
**Pull request by kelsS**
---
name: Add Overlay Features
labels: "feature, ready for review"
---

## pfe-navigation

- add overlay feature that is visible when menu is open and hidden when menu is closed
- add overlay feature that allows user to close entire menu when the overlay is clicked
- insure that aria states toggle appropriately when overlay is clicked to close menu
- overlay is hidden when escape key is pressed and menu closes
- add demo page that shows only one single nav on the page

### Testing instructions

1. Pull down changes from this branch
2. run `npm start`
3. run `npm run dev pfe-navigation`
4. navigate to `http://localhost:8000/elements/pfe-navigation/demo/single-nav.html`
5. Open each dropdown on desktop
6. Close each dropdown on desktop by clicking the overlay div 
7. Open the menu items on mobile
8. Close the menu items on mobile by clicking on the overlay div

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

### CODE REVIEW

File: elements/pfe-navigation/src/pfe-navigation.html [hunk 1]
```diff
@@ -55,5 +55,4 @@
 
 </nav>
 
-<!-- @todo: figure out to do the overlay in shadow DOM -->
-<!-- <div class="pfe-navigation__overlay"></div> -->
+<div class="pfe-navigation__overlay" hidden></div>
```
Comment (wesruv) [inline elements/pfe-navigation/src/pfe-navigation.html:58]: TIL there's an attribute called hidden!

Comment (kelsS) [reply elements/pfe-navigation/src/pfe-navigation.html:58]: @wesruv yeah, I learned about it fairly recently as well! It is a cool attribute.

File: elements/pfe-navigation/src/pfe-navigation.js [hunk 2]
```diff
@@ -1039,6 +1052,12 @@ class PfeNavigation extends PFElement {
     }
   }
 
+  // close menu when overlay is clicked
+  _overlayClickHandler() {
+    const openToggleId = this.getAttribute(`${this.tag}-open-toggle`);
+    this._changeNavigationState(openToggleId, "close");
```
Comment (wesruv) [inline elements/pfe-navigation/src/pfe-navigation.js:1058]: I think this click handler needs to be a little more robust like we talked about
```js
  // close menu when overlay is clicked
  _overlayClickHandler() {
    const openToggleId = this.getAttribute(`${this.tag}-open-toggle`);
    if (this.isSecondaryLinksSectionCollapsed()) {
      // Mobile
      this._changeNavigationState('mobile__button', "close");
    }
    else if (this.isMobileMenuButtonVisible()) {
      // Tablet-ish
      // @todo if it's a child of main menu (e.g. openToggleId.startsWith("main-menu")) close mobile__button
      // Else close openToggleId
    } else {
      // Desktop
      this._changeNavigationState(openToggleId, "close");
    }
  }
```

**Review by wesruv — APPROVED** [review]
It looks good from a code perspective to me, if you're happy with it go ahead and merge!

**Pull request merged**